React/CSS-in-JS/Chakra.md

```diff
@@ -0,0 +1,498 @@
+# Chakra UI
+
+Chakra UI is a simple, modular and accessible component library that gives you the building blocks you need to build your React applications.
+
+## Getting started
+
+### Installation:
+
+```
+npm i @chakra-ui/react @emotion/react @emotion/styled framer-motion
+```
+
+### Setup provider:
+
+For Chakra UI to work correctly, you need to set up the `ChakraProvider` at the root of your application.
+
+Example:
+
+```jsx
+import { ChakraProvider } from "@chakra-ui/react";
+
+function App({ Component, pageProps }: AppProps): ReactElement {
+  return (
+    <ChakraProvider theme={theme}>
+      <Component {...pageProps} />
+    </ChakraProvider>
+  );
+}
+
+export default App;
+```
+
+Note:
+
+- For Next.js, you need to set this up in `pages/_app.tsx`
+- For Create React App, you need to set this up in `index.tsx`
+
```

Comment:
Add a note that ChakraProvider uses emotion ThemeProvider under the hood. 
https://github.com/chakra-ui/chakra-ui/blob/develop/packages/react/src/chakra-provider.tsx#L57
https://github.com/chakra-ui/chakra-ui/blob/develop/packages/system/src/providers.tsx#L12

Mention that it uses [styled-system](https://styled-system.com/) too
https://github.com/chakra-ui/chakra-ui/blob/develop/packages/styled-system/src/system.ts#L1
Explain [responsive-styles](https://styled-system.com/responsive-styles) props 

Pseudo props should also be mentioned somewhere https://chakra-ui.com/docs/features/style-props#pseudo, this is specific to Chakra UI